components/tidb_query/src/util.rs

```diff
@@ -103,9 +105,38 @@ pub fn get_pk(col: &ColumnInfo, h: i64) -> Datum {
     }
 }
 
+#[inline]
+pub fn check_record_key(key: &[u8]) -> Result<(), EvaluateError> {
+    check_key_type(key, table::RECORD_PREFIX_SEP)
+}
+
+#[inline]
+pub fn check_index_key(key: &[u8]) -> Result<(), EvaluateError> {
+    check_key_type(key, table::INDEX_PREFIX_SEP)
+}
+
+/// `check_key_type` checks if the key is the type we want, `wanted_type` should be
+/// `table::RECORD_PREFIX_SEP` or `table::INDEX_PREFIX_SEP` .
+#[inline]
+pub fn check_key_type(key: &[u8], wanted_type: &[u8]) -> Result<(), EvaluateError> {
+    let sep = &key[table::TABLE_PREFIX_KEY_LEN..table::PREFIX_LEN];
+    if wanted_type != sep {
+        Err(EvaluateError::Other(format!(
+            "Expected key type {:?}, but got {:?})",
```

Comment:
prefer to print the whole key so that tracking the wrong row can be easier.

BTW according to our unified key format rfc, the key should be printed using upper hex. https://github.com/tikv/rfcs/blob/master/text/2018-11-12-unified-key-format.md